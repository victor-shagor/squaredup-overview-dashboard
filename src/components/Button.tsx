import styled from "styled-components";
import { ButtonProps } from "constants/interfaces";

const Button = ({ onClick, active, text, testId }: ButtonProps) => {
  return (
    <Styling onClick={onClick} active={active} text={text} data-testid={testId}>
      {text}
    </Styling>
  );
};

const Styling = styled.button<ButtonProps>`
  background-color: ${(props) => (props.active ? "#2C2C2C" : "white")};
  font-family: SFProBold;
  cursor: pointer;
  padding: 15px 30px 15px 30px;
  color: ${(props) => (props.active ? "white" : "#2C2C2C")};
  border-radius: 20px;
  border: none;
`;

export default Button;
