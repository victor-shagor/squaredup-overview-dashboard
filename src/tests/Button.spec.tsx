import { fireEvent, render, screen } from "@testing-library/react";
import Button from "components/Button";
import renderer from "react-test-renderer";

describe("Button Component", () => {
  it("render button component with the correct style when not active", () => {
    const onClick = jest.fn();

    render(
      <Button text={"test"} onClick={onClick} testId="testid" active={false} />
    );
    expect(screen.getByTestId("testid")).toHaveStyle(`background-color:white`);
    expect(screen.getByTestId("testid")).toHaveStyle(`color:#2C2C2C`);
  });

  it("render active button component with the correct style", () => {
    const onClick = jest.fn();

    render(
      <Button text={"test"} onClick={onClick} testId="testid" active={true} />
    );
    expect(screen.getByTestId("testid")).toHaveStyle(
      `background-color:#2C2C2C`
    );
    expect(screen.getByTestId("testid")).toHaveStyle(`color:white`);
  });

  it("onclick event should be triggered", () => {
    const onClick = jest.fn();

    render(
      <Button text={"test"} onClick={onClick} testId="testid" active={true} />
    );
    const buttonClick = screen.getByTestId("testid");
    expect(onClick).toHaveBeenCalledTimes(0);
    fireEvent.click(buttonClick);
    expect(onClick).toHaveBeenCalled();
  });

  it("renders correctly", () => {
    const onClick = jest.fn();
    const tree = renderer
      .create(
        <Button text={"test"} onClick={onClick} testId="testid" active={true} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
