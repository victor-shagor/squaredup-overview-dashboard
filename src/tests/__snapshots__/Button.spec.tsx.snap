// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders correctly 1`] = `
<button
  className="sc-bczRLJ gPslix"
  data-testid="testid"
  onClick={[MockFunction]}
>
  test
</button>
`;
